atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosServerImplementation.java

```diff
@@ -0,0 +1,233 @@
+                ImmutableList.copyOf(acceptors),
+                ImmutableList.copyOf(learners),
+                getQuorumSize(configuration),
+                executor);
+
+        // Build and store a gatekeeping leader election service
+        leaderElectionService = new PaxosLeaderElectionService(
+                proposer,
+                paxosResource.getPaxosLearner(LEADER_NAMESPACE),
+                otherLeaders,
+                ImmutableList.copyOf(acceptors),
+                ImmutableList.copyOf(learners),
+                executor,
+                paxosConfiguration.pingRateMs(),
+                paxosConfiguration.randomWaitBeforeProposalMs(),
+                paxosConfiguration.leaderPingResponseWaitMs());
+        environment.jersey().register(leaderElectionService);
+    }
+
+    private static int getQuorumSize(TimeLockServerConfiguration configuration) {
+        return configuration.cluster().servers().size() / 2 + 1;
+    }
+
+    private static Optional<SSLSocketFactory> constructOptionalSslSocketFactory(
+            PaxosConfiguration configuration) {
+        return Optional.of(SslSocketFactories.createSslSocketFactory(configuration.sslConfiguration().get()));
+    }
+
+    @Override
+    public void onStop() {
+        // Nothing to do
+    }
+
+    @Override
+    public void onStartupFailure() {
+        // Nothing
+    }
+
+    @Override
+    public TimeLockServices createInvalidatingTimeLockServices(String client) {
+        TimestampService timestampService = createPaxosBackedTimestampService(client);
+        LockService lockService = AwaitingLeadershipProxy.newProxyInstance(
+                LockService.class,
+                LockServiceImpl::create,
+                leaderElectionService);
+
+        return TimeLockServices.create(timestampService, lockService);
+    }
+
+    private TimestampService createPaxosBackedTimestampService(String client) {
+        paxosResource.addClient(client);
+
+        ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder()
+                .setNameFormat("atlas-consensus-" + client + "-%d")
+                .setDaemon(true)
+                .build());
+
+        List<PaxosAcceptor> acceptors = getNamespacedProxies(
+                remoteServers,
+                client,
+                optionalSecurity,
+                PaxosAcceptor.class);
+        acceptors.add(paxosResource.getPaxosAcceptor(client));
+        List<PaxosLearner> learners = getNamespacedProxies(
+                remoteServers,
+                client,
+                optionalSecurity,
+                PaxosLearner.class);
+        learners.add(paxosResource.getPaxosLearner(client));
+
+        PaxosProposer proposer = PaxosProposerImpl.newProposer(
+                paxosResource.getPaxosLearner(client),
+                ImmutableList.copyOf(acceptors),
+                ImmutableList.copyOf(learners),
+                acceptors.size() / 2 + 1,
```

Comment:
I'm just trying to figure this out - but is this all cool if there is only one node in the configuration, and it's localhost? As far as I can tell, `acceptors.size() = 0`, and I wasn't sure if that was OK.

In a one-node configuration, are you still constructing a kind of one-node Paxos?